src/adapters/MeetingsAdapter.js

```diff
@@ -0,0 +1,90 @@
+   *
+   * @typedef {Object} Meeting
+   * @property {string}  ID The meeting identifier.
+   * @property {string}  meetingTitle The title of the meeting.
+   * @property {Date}    meetingTimeStart The time and date of the start of the meeting.
+   * @property {Date}    meetingTimeEnd The time and date of the end of the meeting.
+   * @property {Boolean} localAudioMuted If the local user has muted their audio.
+   * @property {Boolean} localVideoMuted If the local user has muted their video.
+   * @property {Boolean} meetingJoined If the local user has joined the meeting.
+   */
+
+  /**
+   * Returns an observable that emits meeting data of the given ID.
+   *
+   * @param {string} ID  ID of the meeting to get.
+   * @returns {Observable.<Meeting>}
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  getMeeting(ID) {
+    return throwError(new Error('getMeeting(ID) must be defined in MeetingsAdapter'));
+  }
+
+  /**
+   * Returns a promise that resolved with the meeting ID that matches
+   * the requested parameters.
+   *
+   * @param {string} destinationID
+   * @param {string} destinationType
+   * @returns {Promise.<string>}
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  getMeetingID(destinationID, destinationType) {
+    return Promise.reject(new Error('getMeetingID(destinationID, destinationType) must be defined in MeetingsAdapter'));
+  }
+
+  /**
+   * Sets the local audio muted state of a particular meetingID.
+   * Returns a promise that is resolved when the set completes.
+   * Also emits an update to the getMeeting observable with the state update to `localAudioMuted`.
+   *
+   * @param {Boolean} audioMuted
+   * @param {string} meetingID
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  setLocalAudioMuted(audioMuted, meetingID) {
+    return Promise.reject(new Error('setLocalAudioMuted(audioMuted, meetingID) must be defined in MeetingsAdapter'));
+  }
+
+  /**
+   * Sets the local video muted state of a particular meetingID.
+   * Returns a promise that is resolved when the set completes.
+   * Also emits an update to the getMeeting observable with the state update to `localVideoMuted`.
+   *
+   * @param {Boolean} videoMuted
+   * @param {string} meetingID
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  setLocalVideoMuted(videoMuted, meetingID) {
+    return Promise.reject(new Error('setLocalAudioMuted(videoMuted, meetingID) must be defined in MeetingsAdapter'));
+  }
+
+  /**
+   * Joins a meeting with the local user.
+   * Returns a promise that is resolved when the set completes.
+   * Also emits an update to the getMeeting observable with the state update to `meetingJoined`.
+   *
+   * @param {string} meetingID
+   * @memberof MeetingsAdapter
+   */
+  // eslint-disable-next-line no-unused-vars
+  joinMeeting(meetingID) {
```

Comment:
I think it would make sense for this to turn the same observables as `getMeeting`, this way, when a component joins the meeting, they also have information about the meeting they joined, and receive updates as thing change.

Reply:
I guess this gets back to how are we going to handle actions in the adapters. Since we're not really expecting a return object here, wouldn't it be more direct to emit the change in the `getMeeting` observable?